conan_provider.cmake

```diff
@@ -247,6 +247,25 @@ function(detect_host_profile output_file)
 
     string(APPEND PROFILE "[conf]\n")
     string(APPEND PROFILE "tools.cmake.cmaketoolchain:generator=${CMAKE_GENERATOR}\n")
+
+    # propagate compilers via profile
+    set(_conan_c_compiler "")
+    set(_conan_cpp_compiler "")
+    if(CMAKE_C_COMPILER)
+        set(_conan_c_compiler "\"c\":\"${CMAKE_C_COMPILER}\",")
```

Comment:
This logic will force all user projects to do C CXX languages or get the warnings? Maybe reduce the messages to STATUS?

Reply:
It doesn't force all users to enable both languages, but it does enforce defining both compilers - there's a distinction there, one can define the `CMAKE_<LANG>_COMPILER` variables without enabling the languages.

While this is unfortunate, I think this is a valid warning, unless we have a different logic and try to guess the other compiler when only one is defined.

If the project only enables C++ and the compiler is clang on Linux - ANY dependency that requires the C language will be built with the system default (gcc). Conan will pass `-stdlib=` and the build of that dependency will fail because gcc does not recognize that option. This does not leave the user with any clue as to what is going on, unfortunately. It think the best course of action is to issue a warning that is more visible than just a status.